index.js

```diff
@@ -121,11 +109,46 @@ var publicAgent = {
       return this;
     }
 
-    var config = initConfig(projectConfig);
+    var logLevelValid = true;
+
+    // Initialize config object
+    var config = {};
+    util._extend(config, require('./config.js').trace);
+    util._extend(config, projectConfig);
+    if (process.env.hasOwnProperty('GCLOUD_TRACE_LOGLEVEL')) {
+      if (process.env.GCLOUD_TRACE_LOGLEVEL.match(/^\d+$/)) {
+        config.logLevel = parseInt(process.env.GCLOUD_TRACE_LOGLEVEL);
+      } else {
+        logLevelValid = false;
+      }
+    }
+    if (process.env.hasOwnProperty('GCLOUD_PROJECT')) {
+      config.projectId = process.env.GCLOUD_PROJECT;
+    }
+
     if (!config.enabled) {
       return this;
     }
-    var logger = common.logger.create(config.logLevel, '@google/cloud-trace');
+
+    var logLevel = config.logLevel;
+    if (logLevel < 0) {
+      logLevel = 0;
+      logLevelValid = false;
+    } else if (logLevel >= common.logger.LEVELS.length) {
+      logLevel = common.logger.LEVELS.length - 1;
+      logLevelValid = false;
+    }
+    var logger = common.logger({
+      level: common.logger.LEVELS[logLevel],
+      tag: '@google/cloud-trace'
+    });
+    if (!logLevelValid) {
+      var desiredLogLevel = process.env.GCLOUD_TRACE_LOGLEVEL ||
+        config.logLevel;
+      logger.warn('Provided log level ' + desiredLogLevel + ' is invalid, ' +
```

Comment:
I think you can use `console.error` here to avoid issues with the default logLevel not allowing `warn` to actually be printed. This would be a valid use of `console.error`.

This should also simplify the logic here as you wouldn't need to carry around the `logLevelValid` flag and you could print the error as you encounter it.

It would be good to get this code back into a initConfig function.